smart-contracts/assembly/contracts/NFT/NFT-internals.ts

```diff
@@ -1,282 +1,239 @@
 import {
   stringToBytes,
-  bytesToU64,
-  u64ToBytes,
-  Args,
-  SafeMath,
+  bytesToU256,
+  bytesToString,
+  boolToByte,
+  byteToBool,
+  u256ToBytes,
 } from '@massalabs/as-types';
-import { Storage, Context, validateAddress } from '@massalabs/massa-as-sdk';
+import { Storage, Context } from '@massalabs/massa-as-sdk';
 
-import {
-  approvedForAllTokenKey,
-  approvedTokenKey,
-  baseURIKey,
-  counterKey,
-  initCounter,
-  nameKey,
-  nft1_ownerOf,
-  ownerKey,
-  ownerTokenKey,
-  symbolKey,
-  totalSupplyKey,
-} from './NFT';
-
-/**
- * Initialize all the properties of the NFT (contract Owner, counter to 0...)
- *
- * @param args - Args object serialized as a string containing:
- * - the token name (string)
- * - the token symbol (string).
- * - the totalSupply (u64)
- * - the baseURI (string)
- */
-export function _constructor(args: Args): void {
-  // This line is important. It ensures that this function can't be called in the future.
-  // If you remove this check, someone could call your constructor function and reset your smart contract.
-  assert(Context.isDeployingContract());
+import { u256 } from 'as-bignum/assembly';
 
-  const name = args.nextString().expect('name argument is missing or invalid');
-  const symbol = args
-    .nextString()
-    .expect('symbol argument is missing or invalid');
-  const totalSupply = args
-    .nextU64()
-    .expect('totalSupply argument is missing or invalid');
-  const baseURI = args
-    .nextString()
-    .expect('baseURI argument is missing or invalid');
+export const NAME_KEY: StaticArray<u8> = [0x01];
+export const SYMBOL_KEY: StaticArray<u8> = [0x02];
 
-  Storage.set(nameKey, name);
-  Storage.set(symbolKey, symbol);
-  Storage.set(totalSupplyKey, u64ToBytes(totalSupply));
-  Storage.set(baseURIKey, baseURI);
-  Storage.set(ownerKey, Context.caller().toString());
-  Storage.set(counterKey, u64ToBytes(initCounter));
-}
+export const BALANCE_KEY_PREFIX: StaticArray<u8> = [0x03];
+export const OWNER_KEY_PREFIX: StaticArray<u8> = [0x04];
+export const ALLOWANCE_KEY_PREFIX: StaticArray<u8> = [0x05];
+export const OPERATOR_ALLOWANCE_KEY_PREFIX: StaticArray<u8> = [0x06];
```

Comment:
Personal preference: I would use an enum instead. SOmething like:
```suggestion
export enum Key {
  NAME = [0x01],
  SYMBOL = [0x02],
  BALANCE_PREFIX = [0x03],
  OWNER_PREFIX = [0x04],
  ALLOWANCE_PREFIX = [0x05],
  OPERATOR_ALLOWANCE_PREFIX = [0x06]
}
```